src/Datadog.Trace/TraceContext.cs

```diff
@@ -98,16 +97,16 @@ public void CloseSpan(Span span)
                 }
             }
 
-            List<Span> spansToWrite = null;
+            Span[] spansToWrite = null;
 
-            lock (_lock)
+            lock (_spans)
             {
                 _openSpans--;
 
                 if (_openSpans == 0)
                 {
-                    spansToWrite = _spans;
-                    _spans = new List<Span>();
+                    spansToWrite = _spans.ToArray();
+                    _spans.Clear();
```

Comment:
What do you think about resetting the capacity down to the default level or a smaller level after performing this Clear operation, instead of keeping the existing capacity?

Reply:
The intent here was to keep the list's current internal array instead of throwing it away, in case more spans are added after this `Clear()`.

If we trim this list down to zero capacity (the default), it means a new array of size 4 is allocated if another span is added.

I don't think it will make a big difference either way. This should rarely happen since it means adding a span _after_ all other spans (including the root) have finished.